mailpoet/lib/Automation/Engine/Control/StepHandler.php

```diff
@@ -108,7 +120,22 @@ private function handleStep(array $args): void {
 
     $stepType = $step->getType();
     if (isset($this->stepRunners[$stepType])) {
-      $this->stepRunners[$stepType]->run($step, $workflow, $workflowRun);
+      $log = new WorkflowRunLog($workflowRun->getId(), $step->getId(), $args);
+      try {
+        $this->stepRunners[$stepType]->run($step, $workflow, $workflowRun);
+        $log->markCompletedSuccessfully();
+      } catch (Throwable $e) {
+        $log->markFailed();
+        $log->setError($e);
+        throw $e;
+      } finally {
+        try {
+          $this->hooks->doWorkflowStepAfterRun($log);
+        } catch (Throwable $e) {
+          // Ignore integration errors
```

Comment:
I'm not sure if simply ignoring errors during the hook is what we want since it could make debugging very difficult for integration authors. @JanJakes can you think of a better way we could handle this, or does it seem fine for now?